mysql-test/suite/period/t/overlaps.test

```diff
@@ -0,0 +1,139 @@
+                          period for p(s,e),
+                          primary key(id, q without overlaps));
+
+--error ER_PERIOD_NOT_FOUND
+create or replace table t(id int, s date, e date,
+                          primary key(id, p without overlaps));
+
+--error ER_KEY_CONTAINS_PERIOD_FIELDS
+create or replace table t(id int, s date, e date,
+                          period for p(s,e),
+                          primary key(id, s, p without overlaps));
+
+create or replace table t(id int, s date, e date,
+                          period for p(s,e),
+                          primary key(id));
+insert into t values (1, '2003-03-01', '2003-05-01');
+--error ER_DUP_ENTRY
+insert into t values (1, '2003-04-01', '2003-05-01');
+
+create or replace table t(id int, u int, s date, e date,
+                          period for p(s,e),
+                          primary key(id, p without overlaps),
+                          unique(u));
+show create table t;
+insert into t values (1, 1, '2003-03-01', '2003-05-01');
+insert into t values (1, 2, '2003-05-01', '2003-07-01');
+--error ER_DUP_KEY
+insert into t values (1, 3, '2003-04-01', '2003-05-01');
+
+create or replace table t(id int, u int, s date, e date,
+                          period for p(s,e),
+                          primary key(id, p without overlaps),
+                          unique(u, p without overlaps));
+show create table t;
+insert into t values (1, 1, '2003-03-01', '2003-05-01');
+insert into t values (1, 2, '2003-05-01', '2003-07-01');
+insert into t values (2, 1, '2003-05-01', '2003-07-01');
+
+create or replace table t(id int, s date, e date,
+                          period for p(s,e)) engine=innodb;
+
+insert into t values (1, '2003-01-01', '2003-03-01'),
+                     (1, '2003-05-01', '2003-07-01'),
+                     (1, '2003-02-01', '2003-04-01');
+
+--replace_regex /#sql-\w+/#sql-temp/
+--error ER_DUP_KEY
+alter table t add primary key(id, p without overlaps);
+
+--echo # Historical rows are not checked against constraints
+set @@system_versioning_alter_history= keep;
+alter table t add system versioning;
+delete from t;
+alter table t add primary key(id, p without overlaps);
+
+insert into t values (1, '2003-01-01', '2003-03-01'),
+                     (1, '2003-03-01', '2003-05-01');
+
+--error ER_PERIOD_WITHOUT_OVERLAPS_PARTITIONED
+create or replace table t1 (x int, s date, e date,
+                            period for p(s,e),
+                            primary key(x, p without overlaps)
+) with system versioning
+  partition by key (x);
+
+--error ER_PERIOD_WITHOUT_OVERLAPS_PARTITIONED
+create or replace table t1 (x int, s date, e date,
+                            period for p(s,e),
+                            primary key(x, p without overlaps)
+) with system versioning
+  partition by system_time limit 2
+  (partition p0 history,
+   partition p1 history,
+   partition pn current);
+
+create or replace database test;
```

Comment:
## Bug 4: non-unique key is allowed and causes segfault
### Reproduce
```sql
create or replace table t2 (x int, s date, e date, period for p (s, e), key(x, p without overlaps));
show create table t2;
```
### Result
```c++
Thread 33 "mysqld" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7ffff4222700 (LWP 14964)]
0x000000000084e4c5 in show_create_table (thd=0x7fff88000cf8, table_list=0x7fff88015098, packet=0x7ffff421db90, create_info_arg=0x0, with_db_name=WITHOUT_DB_NAME) at /home/midenok/src/mariadb/review/src/sql/sql_show.cc:2355
2355          if (field->invisible > INVISIBLE_USER)
(gdb) bt
#0  0x000000000084e4c5 in show_create_table (thd=0x7fff88000cf8, table_list=0x7fff88015098, packet=0x7ffff421db90, create_info_arg=0x0, with_db_name=WITHOUT_DB_NAME) at /home/midenok/src/mariadb/review/src/sql/sql_show.cc:2355
#1  0x000000000084c2e4 in mysqld_show_create_get_fields (thd=0x7fff88000cf8, table_list=0x7fff88015098, field_list=0x7ffff421db78, buffer=0x7ffff421db90) at /home/midenok/src/mariadb/review/src/sql/sql_show.cc:1292
#2  0x000000000084f22b in mysqld_show_create (thd=0x7fff88000cf8, table_list=0x7fff88015098) at /home/midenok/src/mariadb/review/src/sql/sql_show.cc:1366
#3  0x000000000079efdd in mysql_execute_command (thd=0x7fff88000cf8) at /home/midenok/src/mariadb/review/src/sql/sql_parse.cc:4535
#4  0x0000000000797510 in mysql_parse (thd=0x7fff88000cf8, rawbuf=0x7fff88014fc0 "show create table t2", length=20, parser_state=0x7ffff4221640, is_com_multi=false, is_next_command=false) at /home/midenok/src/mariadb/review/src/sql/sql_parse.cc:8078
```
### Expected
Error message that non-unique key is not allowed for `WITHOUT OVERLAPS`.

Reply:
fixed